crates/accelerate/src/twirling.rs

```diff
@@ -233,15 +228,48 @@ fn generate_twirled_circuit(
     custom_gate_map: Option<&CustomGateTwirlingMap>,
     optimizer_target: Option<&Target>,
 ) -> PyResult<CircuitData> {
-    let mut out_circ = CircuitData::copy_empty_like(circ, VarsMode::Alike)?;
+    let mut out_circ = CircuitData::copy_empty_like(circ, VarsMode::Alike, false)?;
 
-    for inst in circ.data() {
+    for (index, inst) in circ.data().iter().enumerate() {
```

Comment:
The need to include the enumeration here feels like it'll cause API problems in the future, because it implies that `(&CircuitData, &PackedInstruction)` is no longer enough to get all the information out - you also now need the index, which is very new for `CircuitData` in particular.

I think the main comment on these will be on `CircuitData`/`DAGCircuit`, though, and this is just an instance of it.